### ISSUE TITLE
Incorrect results when converting SVG to PNG

### ISSUE THREAD

**Issue by qarmin** [thorvg/thorvg#1209]
![Screenshot from 2022-04-03 10-44-56](https://user-images.githubusercontent.com/41945903/161419682-224a2216-908b-4b82-a93b-9ee980e79c20.png)

[Example.zip](https://github.com/Samsung/thorvg/files/8404102/Example.zip)

**Comment by mgrudzinska** [comment]
2224 - text is not supported 
2229 - after #1214 it's better, but not perfect, since tvg doesn't support filters
2684 - filters are not supported

**Comment by hermet** [comment]
1550, 1552, 2371, 2909 - confirm fixed
2192, 3323 - gradient filling of body is a bit different
2224, 2229, 2684 - no plan to support it yet

**Comment by hermet** [comment]
@qarmin fixed 1550, 1552, 2371, 2909 in https://github.com/Samsung/thorvg/releases/tag/v0.8.1

**Comment by JSUYA** [comment]
> 2224, 2229, 2684 - no plan to support it yet

details
2224 - text
2229 - feColorMatrix filter
2684 - feGaussianBlur filter

+) Confirm fixed with #1213

**Comment by hermet** [comment]
Still need to check:
2192, 3323 - gradient filling of body is a bit different

**Comment by capnm** [comment]
2961 fixed

**Comment by hermet** [comment]
@capnm thanks for confirm it!

**Comment by capnm** [comment]
@hermet 
2192 - has still this gradient-inverted issue

**Comment by hermet** [comment]
Yes, I just checked. it. Thanks.

**Comment by hermet** [comment]
2192 test result

![image](https://user-images.githubusercontent.com/3711518/232265732-8253dd84-58b7-4f62-a7d7-58bfde3e0734.png)

@capnm tvg & major browsers have the same result.

**Comment by hermet** [comment]
Created a separate issue for the filter enhancement : https://github.com/thorvg/thorvg/issues/1367

**Comment by capnm** [comment]
> Ah confused. 2192 is also fixed in the master branch.

The _gradientTransform="matrix(0.5,0,0,-0.48276,-69.03301,54.17094)"_ seems to be here not supported?

```svg
<linearGradient
       id="a-3"
       y1="68.419998"
       y2="-3.454"
       x1="3.1159999"
       x2="61.688999"
       gradientUnits="userSpaceOnUse"
       gradientTransform="matrix(0.5,0,0,-0.48276,-69.03301,54.17094)">
      <stop
         stop-color="#66b036"
         id="stop7-6" />
      <stop
         offset="1"
         stop-color="#239161"
         id="stop9-7" />
    </linearGradient>
```

**Comment by mgrudzinska** [comment]
@capnm please give more details, example svg. gradientTransform is supported.
from [this](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/gradientTransform) sample skew is not supported, but all the samples from this issue seems to be ok, don't they?

**Comment by capnm** [comment]
> @capnm please give more details, example svg. gradientTransform is supported. from [this](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/gradientTransform) sample skew is not supported, but all the samples from this issue seems to be ok, don't they?

The gradientTransform is inside 2192, as mentioned above. I still see it lighter at the top. https://github.com/thorvg/thorvg/issues/1209#issuecomment-1498553601 

![grafik](https://user-images.githubusercontent.com/4047289/232304385-59fa72d0-5c37-472b-8c64-abc9f47a82a6.png)
2192.svg:
![2192](https://user-images.githubusercontent.com/4047289/232304459-49ec4101-1d4a-49fb-94a8-34918f0c4b6a.svg)

**Comment by mgrudzinska** [comment]
@capnm please use the newest master:
<img width="376" alt="Zrzut ekranu 2023-04-16 o 13 16 32" src="https://user-images.githubusercontent.com/67589014/232305758-1933ea6a-8215-48b1-8870-8fe2b008bc98.png">
this was fixed in #1356

**Comment by capnm** [comment]
Oh I see In the mozilla example isn't also   _transform: translateX(220px);_ in tvg  working ...
I compiled Godot with the current tip, your PR change is inside.

![grafik](https://user-images.githubusercontent.com/4047289/232307012-a0771fc2-5444-430b-959f-e0d813c7ff97.png)

**Comment by capnm** [comment]
E.g. 
<details><summary>Godot Quit deadlock</summary>


```
^C
Thread 1 "godot.linuxbsd." received signal SIGINT, Interrupt.
__pthread_clockjoin_ex (threadid=140736857073408, thread_return=0x0, clockid=<optimised out>, abstime=<optimised out>, block=<optimised out>) at pthread_join_common.c:145
145	pthread_join_common.c: No such file or directory.
(gdb) bt
#0  __pthread_clockjoin_ex (threadid=140736857073408, thread_return=0x0, clockid=<optimised out>, abstime=<optimised out>, block=<optimised out>) at pthread_join_common.c:145
#1  0x000055555a8f8ab7 in std::thread::join() ()
#2  0x00005555575879c8 in tvg::TaskSchedulerImpl::~TaskSchedulerImpl (this=0x55555fd73760) at thirdparty/thorvg/src/lib/tvgTaskScheduler.cpp:120
#3  0x0000555557587755 in tvg::TaskScheduler::term () at thirdparty/thorvg/src/lib/tvgTaskScheduler.cpp:177
#4  0x0000555557586da3 in tvg::Initializer::term (engine=<optimised out>) at thirdparty/thorvg/src/lib/tvgInitializer.cpp:142
#5  0x0000555557586b05 in uninitialize_svg_module (p_level=p_level@entry=MODULE_INITIALIZATION_LEVEL_SCENE) at modules/svg/register_types.cpp:65
#6  0x0000555556ab7ec1 in uninitialize_modules (p_level=p_level@entry=MODULE_INITIALIZATION_LEVEL_SCENE) at modules/register_module_types.gen.cpp:306
#7  0x0000555556aa80b5 in Main::cleanup (p_force=<optimised out>) at main/main.cpp:3344
#8  0x0000555556a1ca62 in main (argc=<optimised out>, argv=<optimised out>) at platform/linuxbsd/godot_linuxbsd.cpp:75
```
</details>
<details><summary>Crash at start</summary>

```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7c5a859 in __GI_abort () at abort.c:79
#2  0x00007ffff7cc526e in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7def298 "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007ffff7ccd2fc in malloc_printerr (str=str@entry=0x7ffff7ded44d "corrupted size vs. prev_size") at malloc.c:5347
#4  0x00007ffff7ccd96b in unlink_chunk (p=p@entry=0x55555f805570, av=0x7ffff7e24b80 <main_arena>) at malloc.c:1454
#5  0x00007ffff7ccee8b in _int_free (av=av@entry=0x7ffff7e24b80 <main_arena>, p=p@entry=0x55555f8054b0, have_lock=<optimised out>, have_lock@entry=1) at malloc.c:4342
#6  0x00007ffff7cd0f85 in _int_realloc (av=av@entry=0x7ffff7e24b80 <main_arena>, oldp=oldp@entry=0x55555f8054b0, oldsize=oldsize@entry=192, nb=208) at malloc.c:4618
#7  0x00007ffff7cd3156 in __GI___libc_realloc (oldmem=0x55555f8054c0, bytes=188) at malloc.c:3235
#8  0x000055555759bfdc in _growOutlineContour (outline=..., n=1) at thirdparty/thorvg/src/lib/sw_engine/tvgSwShape.cpp:68
#9  _outlineMoveTo (outline=..., to=to@entry=0x55555f8adf60, transform=transform@entry=0x7fffac0042d0) at thirdparty/thorvg/src/lib/sw_engine/tvgSwShape.cpp:115
#10 0x000055555759b4fe in _genOutline (shape=shape@entry=0x55555f99d768, rshape=0x55555f8a12a0, transform=0x7fffac0042d0, mpool=<optimised out>, tid=<optimised out>, hasComposite=false)
    at thirdparty/thorvg/src/lib/sw_engine/tvgSwShape.cpp:450
#11 0x000055555759b294 in shapePrepare (shape=0x2, shape@entry=0x55555f99d768, rshape=0x7fffffffb440, transform=0x0, clipRegion=..., renderRegion=..., mpool=0x0, tid=0, hasComposite=<optimised out>)
    at thirdparty/thorvg/src/lib/sw_engine/tvgSwShape.cpp:484
#12 0x0000555557599826 in SwShapeTask::run (this=0x55555f99d6d0, tid=<optimised out>) at thirdparty/thorvg/src/lib/sw_engine/tvgSwRenderer.cpp:105
#13 0x0000555557598818 in tvg::Task::done (this=0x55555f99d6d0, tid=0) at thirdparty/thorvg/src/lib/tvgTaskScheduler.h:63
#14 tvg::SwRenderer::renderShape (this=0x55555f8aaca0, data=0x55555f99d6d0) at thirdparty/thorvg/src/lib/sw_engine/tvgSwRenderer.cpp:386
#15 0x00005555575a0883 in tvg::Paint::Impl::render (this=0x55555f998b40, renderer=...) at thirdparty/thorvg/src/lib/tvgPaint.cpp:177
#16 0x00005555575a35b1 in tvg::Scene::Impl::render (this=0x55555f9959f0, renderer=...) at thirdparty/thorvg/src/lib/tvgSceneImpl.h:134
#17 0x00005555575a0883 in tvg::Paint::Impl::render (this=0x55555f9be490, renderer=...) at thirdparty/thorvg/src/lib/tvgPaint.cpp:177
#18 0x00005555575a35b1 in tvg::Scene::Impl::render (this=0x55555f9b9cf0, renderer=...) at thirdparty/thorvg/src/lib/tvgSceneImpl.h:134
#19 0x00005555575a0883 in tvg::Paint::Impl::render (this=0x55555f93ce30, renderer=...) at thirdparty/thorvg/src/lib/tvgPaint.cpp:177
#20 0x00005555575a35b1 in tvg::Scene::Impl::render (this=0x55555f987bd0, renderer=...) at thirdparty/thorvg/src/lib/tvgSceneImpl.h:134
#21 0x00005555575a0883 in tvg::Paint::Impl::render (this=0x55555f9b5850, renderer=...) at thirdparty/thorvg/src/lib/tvgPaint.cpp:177
#22 0x00005555575a35b1 in tvg::Scene::Impl::render (this=0x55555f9e2fa0, renderer=...) at thirdparty/thorvg/src/lib/tvgSceneImpl.h:134
#23 0x00005555575a0883 in tvg::Paint::Impl::render (this=0x55555f9c71a0, renderer=...) at thirdparty/thorvg/src/lib/tvgPaint.cpp:177
#24 0x00005555575a0883 in tvg::Paint::Impl::render (this=0x55555f9d7a70, renderer=...) at thirdparty/thorvg/src/lib/tvgPaint.cpp:177
#25 0x000055555759f4c0 in tvg::Canvas::Impl::draw (this=0x55555f9e5f70) at thirdparty/thorvg/src/lib/tvgCanvasImpl.h:120
#26 tvg::Canvas::draw (this=<optimised out>) at thirdparty/thorvg/src/lib/tvgCanvas.cpp:61
#27 0x0000555557596d96 in ImageLoaderSVG::create_image_from_utf8_buffer (this=<optimised out>, p_image=..., p_buffer=..., p_scale=<error reading variable: That operation is not available on integers of more than 8 bytes.>, 
    p_scale@entry=1, p_upsample=<optimised out>) at modules/svg/image_loader_svg.cpp:112
#28 0x0000555557597157 in ImageLoaderSVG::create_image_from_string (this=<optimised out>, p_image=..., p_string=..., p_scale=<error reading variable: That operation is not available on integers of more than 8 bytes.>, 
    p_upsample=<optimised out>, p_color_map=...) at modules/svg/image_loader_svg.cpp:154
#29 0x0000555557b3bebc in editor_generate_icon (p_index=p_index@entry=80, p_scale=<error reading variable: That operation is not available on integers of more than 8 bytes.>, 
    p_saturation=<error reading variable: That operation is not available on integers of more than 8 bytes.>, p_saturation@entry=1, p_convert_colors=...) at editor/editor_themes.cpp:244
#30 0x0000555557b3a91e in editor_register_and_generate_icons (p_theme=..., p_dark_theme=<optimised out>, p_icon_saturation=<error reading variable: That operation is not available on integers of more than 8 bytes.>, 
    p_icon_saturation@entry=1, p_thumb_size=p_thumb_size@entry=64, p_only_thumbs=<optimised out>) at editor/editor_themes.cpp:328
#31 0x0000555557b3f882 in create_editor_theme (p_theme=...) at editor/editor_themes.cpp:632
#32 0x0000555557b7ba07 in create_custom_theme (p_theme=...) at editor/editor_themes.cpp:2044
#33 0x0000555557a07da1 in EditorNode::EditorNode (this=0x55555f921400) at editor/editor_node.cpp:6866
#34 0x0000555556aa43a9 in Main::start () at main/main.cpp:2817
#35 0x0000555556a1ca41 in main (argc=<optimised out>, argv=<optimised out>) at platform/linuxbsd/godot_linuxbsd.cpp:71
```
</details>

**Comment by qarmin** [comment]
Looks that crashes may be related to #1370
